rootfs-builder/gentoo/rootfs_lib.sh

```diff
@@ -0,0 +1,148 @@
+# - Arguments
+#
+# Copyright (c) 2020 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+#
+# rootfs_dir=$1
+#
+# - Optional environment variables
+#
+# EXTRA_PKGS: Variable to add extra PKGS provided by the user
+#
+# BIN_AGENT: Name of the Kata-Agent binary
+#
+# REPO_URL: URL to distribution repository ( should be configured in
+#			config.sh file)
+#
+# Any other configuration variable for a specific distro must be added
+# and documented on its own config.sh
+#
+# - Expected result
+#
+# rootfs_dir populated with rootfs pkgs
+# It must provide a binary in /sbin/init
+#
+gentoo_portage_container=gentoo_portage
+gentoo_local_portage_dir="${HOME}/gentoo-$(date +%s)"
+
+build_rootfs() {
+	# Mandatory
+	local ROOTFS_DIR=$1
+
+	# In case of support EXTRA packages, use it to allow
+	# users to add more packages to the base rootfs
+	local EXTRA_PKGS=${EXTRA_PKGS:-}
+
+	# Populate ROOTFS_DIR
+	# Must provide /sbin/init and /bin/${BIN_AGENT}
+	check_root
+	mkdir -p "${ROOTFS_DIR}"
+
+	# trim whitespace
+	PACKAGES=$(echo $PACKAGES |xargs )
+	EXTRA_PKGS=$(echo $EXTRA_PKGS |xargs)
+
+	# extra packages are added to packages and finally passed to debootstrap
+	if [ "${EXTRA_PKGS}" = ""  ]; then
+		echo "no extra packages"
+	else
+		PACKAGES="${PACKAGES} ${EXTRA_PKGS}"
+	fi
+
+	local packageuseconf="/etc/portage/package.use/user"
+	local makeconf="/etc/portage/make.conf"
+	local systemd_optimizations="acl -apparmor -audit cgroup-hybrid \
+-cryptsetup -curl -dns-over-tls -gcrypt -gnuefi -homed -http \
+-hwdb -idn -importd kmod -lz4 -lzma -nat -pkcs11 -policykit \
+-pwquality -qrcode -repart -resolvconf sysv-utils -test -xkb -zstd"
+	local packages_optimizations="systemd multicall -ncurses -udev -debug \
+-X -selinux -doc -examples -nls -unicode -abi_x86_32 -abi_x86_x32"
+	local compiler_optimizations="-O3 -feliminate-unused-debug-types -pipe \
+-mtune=skylake -fexceptions -fstack-protector --param=ssp-buffer-size=32 \
+-Wl,--copy-dt-needed-entries -m64 -fasynchronous-unwind-tables \
+-Wp,-D_REENTRANT -ftree-loop-distribute-patterns -Wl,-z -Wl,now \
+-Wl,-z -Wl,relro -fno-semantic-interposition -ffat-lto-objects \
+-fno-signed-zeros -fno-trapping-math -fassociative-math -Wl,-sort-common \
+-Wl,--enable-new-dtags"
```

Comment:
Could you build this up line by line using an array or similar? That would make it much clearer, allow comments on any option and make diffs easier to read.